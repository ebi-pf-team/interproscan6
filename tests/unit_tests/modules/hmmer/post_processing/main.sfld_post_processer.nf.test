nextflow_process {

    name "Test Process SFLD_POST_PROCESSER"
    script "interproscan/modules/hmmer/post_processing/main.nf"
    process "SFLD_POST_PROCESSER"

    stage {
        symlink "interproscan/bin/sfld/sfld_postprocess"
        symlink "tests/unit_tests/test_inputs/hmmer_post_processing/sfld/sfld_alignment"
        symlink "tests/unit_tests/test_inputs/hmmer_post_processing/sfld/sfld_dtbl"
        symlink "tests/unit_tests/test_inputs/hmmer_post_processing/sfld/sfld_out"
        symlink "tests/unit_tests/test_inputs/hmmer_post_processing/sfld/sfld_site_anno"
    }

    test("Should run without failures") {

        when {
            params {}
            process {
                """
                input[0] = file("meta/tests/unit_tests/test_inputs/hmmer_post_processing/sfld/sfld_out")
                input[1] = [
                    "../../../meta/tests/unit_tests/test_inputs/hmmer_post_processing/sfld/sfld_site_anno"
                ]
                input[2] = "sfld"
                input[3] = file("meta/tests/unit_tests/test_inputs/hmmer_post_processing/sfld/sfld_alignment")
                input[4] = file("meta/tests/unit_tests/test_inputs/hmmer_post_processing/sfld/sfld_dtbl")
                input[5] = true
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
