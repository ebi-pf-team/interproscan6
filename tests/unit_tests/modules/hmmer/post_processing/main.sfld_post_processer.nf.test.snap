{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "sfld_out.processed.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-09-23T10:12:53.071585"
    }
}